src/object/endpoint.c

```diff
@@ -446,7 +450,7 @@ void cancelBadgedSends(endpoint_t *epptr, word_t badge)
             next = thread->tcbEPNext;
 #ifdef CONFIG_KERNEL_MCS
             /* senders do not have reply objects in their state, and we are only cancelling sends */
-            assert(REPLY_PTR(thread_state_get_replyObject(thread->tcbState)) == NULL);
+            assert(thread_state_get_tsType(thread->tcbState) == ThreadState_BlockedOnSend);
```

Comment:
Is the comment  here, or at least its first part, still something we would like to say?

Reply:
I think so, yes. The first part is arguing that we don't need to care about replyObject part of the thread state (because it is BlockedOnSend). This still applies without the invariant. Before it was arguing that the replyObject is NULL, but the same text still makes sense, I think.